morpheus/_lib/src/io/loaders/rest.cpp

```diff
@@ -17,21 +17,378 @@
+        if (dataframe.is_none())
+        {
+            dataframe = current_df;
+        }
+        else if (strategy == "aggregate")
+        {
+            py::list args;
+            args.attr("append")(dataframe);
+            args.attr("append")(current_df);
+            dataframe = mod_cudf.attr("concat")(args);
+        }
+    }  // release GIL
+}
+
+void create_dataframe_from_query(py::object& dataframe,
+                                 py::module_& mod_cudf,
+                                 nlohmann::json& query,
+                                 int max_retry,
+                                 int retry_interval_milliseconds,
+                                 std::string& strategy)
+{
+    std::string method;
+    std::string endpoint;
+    std::string content_type;
+    std::string body;
+    nlohmann::json params;
+    nlohmann::json x_headers;
+    extract_query_fields(query, method, endpoint, content_type, body, params, x_headers);
+
+    if (params.empty())
+    {
+        http::response<http::dynamic_body> response;
+        get_response_from_endpoint(
+            response, method, endpoint, content_type, body, params, x_headers, max_retry, retry_interval_milliseconds);
+        create_dataframe_from_response(dataframe, mod_cudf, response, strategy);
+    }
+    else
+    {
+        // For each set of param, send a separate request
+        for (auto& param : params)
+        {
+            http::response<http::dynamic_body> response;
+            get_response_from_endpoint(response,
+                                       method,
+                                       endpoint,
+                                       content_type,
+                                       body,
+                                       param,
+                                       x_headers,
+                                       max_retry,
+                                       retry_interval_milliseconds);
+            create_dataframe_from_response(dataframe, mod_cudf, response, strategy);
+        }
+    }
+}
+
+void process_failures(const std::string& error_msg,
+                      std::shared_ptr<ControlMessage> message,
+                      bool processes_failures_as_errors)
+{
+    if (processes_failures_as_errors)
+    {
+        throw std::runtime_error(error_msg);
+    }
+    message->set_metadata("cm_failed", "true");
+    message->set_metadata("cm_failed_reason", error_msg);
+}
+
 std::shared_ptr<ControlMessage> RESTDataLoader::load(std::shared_ptr<ControlMessage> message, nlohmann::json task)
 {
     VLOG(30) << "Called RESTDataLoader::load()";
 
-    // TODO(Devin): Implement this
-    throw std::runtime_error("RESTDataLoader::load() not implemented yet");
+    // If set to false, any exception thrown during the task is caught and the related fields in ControlMessage are set
+    // to indicate the reason of that failure; Otherwise, the exception is thrown
+    bool processes_failures_as_errors = false;
```

Comment:
Could you actually extract this capture process to the base DataLoader implementation in `data_loader.cpp`, then this can form the basis for gracefully handling all loader failures instead of just this one.

std::shared_ptr<ControlMessage> DataLoader::load(std::shared_ptr<ControlMessage> control_message)